### ISSUE TITLE
Problem with testing

### ISSUE THREAD

**Issue by gsagnol** [eendebakpt/oapackage#95]
Here is the output of the following commands on my PC:

    $ python -c "import numpy; print(numpy); print(numpy.__version__)"
    <module 'numpy' from '/homes/-hidden-/.local/lib/python2.7/site-packages/numpy/__init__.pyc'>
    1.15.4

    $ python -c "import oapackage; print(oapackage)"
    <module 'oapackage' from '/homes/-hidden-/.local/lib/python2.7/site-packages/oapackage/__init__.pyc'>

    $ pytest -v
============================= test session starts ==============================
platform linux -- Python 3.6.5, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python3
cachedir: .cache
rootdir: /homes/-hidden-/.local/lib/python2.7/site-packages/oapackage, inifile:
collected 0 items / 2 errors                                                   

==================================== ERRORS ====================================
__________________ ERROR collecting tests/test_cpplibrary.py ___________________
ImportError while importing test module '/homes/-hidden-/.local/lib/python2.7/site-packages/oapackage/tests/test_cpplibrary.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../oalib.py:17: in swig_import_helper
    return importlib.import_module(mname)
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:658: in _load_unlocked
    ???
<frozen importlib._bootstrap>:571: in module_from_spec
    ???
<frozen importlib._bootstrap_external>:922: in create_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
E   ImportError: dynamic module does not define module export function (PyInit__oalib)

During handling of the above exception, another exception occurred:
__init__.py:10: in <module>
    import oalib
../oalib.py:20: in <module>
    _oalib = swig_import_helper()
../oalib.py:19: in swig_import_helper
    return importlib.import_module('_oalib')
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ImportError: dynamic module does not define module export function (PyInit__oalib)
___________________ ERROR collecting tests/test_oapackage.py ___________________
ImportError while importing test module '/homes/-hidden-/.local/lib/python2.7/site-packages/oapackage/tests/test_oapackage.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../oalib.py:17: in swig_import_helper
    return importlib.import_module(mname)
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:994: in _gcd_import
    ???
<frozen importlib._bootstrap>:971: in _find_and_load
    ???
<frozen importlib._bootstrap>:955: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:658: in _load_unlocked
    ???
<frozen importlib._bootstrap>:571: in module_from_spec
    ???
<frozen importlib._bootstrap_external>:922: in create_module
    ???
<frozen importlib._bootstrap>:219: in _call_with_frames_removed
    ???
E   ImportError: dynamic module does not define module export function (PyInit__oalib)

During handling of the above exception, another exception occurred:
__init__.py:10: in <module>
    import oalib
../oalib.py:20: in <module>
    _oalib = swig_import_helper()
../oalib.py:19: in swig_import_helper
    return importlib.import_module('_oalib')
/usr/lib64/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
E   ImportError: dynamic module does not define module export function (PyInit__oalib)
!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 2 error in 0.57 seconds ============================

**Comment by gsagnol** [comment]
I also installed `OApackage` on my laptop (with a global installation), and this time the tests work, but with several failures:

    $ pytest -v

==================================================== test session starts =====================================================
platform linux2 -- Python 2.7.15rc1, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /usr/bin/python2
cachedir: .cache
rootdir: /usr/local/lib/python2.7/dist-packages/oapackage, inifile:
collected 70 items                                                                                                           

tests/test_cpplibrary.py::TestReductions::test_DOP PASSED                                                              [  1%]
tests/test_cpplibrary.py::TestReductions::test_LMC PASSED                                                              [  2%]
tests/test_cpplibrary.py::TestModelmatrix::test_modelmatrix PASSED                                                     [  4%]
tests/test_cpplibrary.py::TestArrayLink::test_array_class_functions PASSED                                             [  5%]
tests/test_cpplibrary.py::TestArrayLink::test_array_link_dimensions PASSED                                             [  7%]
tests/test_cpplibrary.py::TestArrayLink::test_basic_array_link_functionality PASSED                                    [  8%]
tests/test_cpplibrary.py::TestArrayLink::test_is_ortogonal_array PASSED                                                [ 10%]
tests/test_cpplibrary.py::TestArrayLink::test_selectFirstColumns PASSED                                                [ 11%]
tests/test_cpplibrary.py::TestArraydata_t::test_factor_levels PASSED                                                   [ 12%]
tests/test_cpplibrary.py::TestConferenceDesigns::test_conf2dsd PASSED                                                  [ 14%]
tests/test_cpplibrary.py::TestArrayFiles::test_write_latex_format PASSED                                               [ 15%]
tests/test_cpplibrary.py::TestCppLibrary::test_array2eigenModelMatrixMixed PASSED                                      [ 17%]
tests/test_cpplibrary.py::TestCppLibrary::test_array_transformation_t PASSED                                           [ 18%]
tests/test_cpplibrary.py::TestCppLibrary::test_arraylink PASSED                                                        [ 20%]
tests/test_cpplibrary.py::TestCppLibrary::test_arraylink2arraydata PASSED                                              [ 21%]
tests/test_cpplibrary.py::TestCppLibrary::test_arraylink_slicing PASSED                                                [ 22%]
tests/test_cpplibrary.py::TestCppLibrary::test_arrayrankInfo PASSED                                                    [ 24%]
tests/test_cpplibrary.py::TestCppLibrary::test_conference_generation PASSED                                            [ 25%]
tests/test_cpplibrary.py::TestCppLibrary::test_exception_handling hiFAILED                                               [ 27%]
tests/test_cpplibrary.py::TestCppLibrary::test_mvalue_t PASSED                                                         [ 28%]
tests/test_cpplibrary.py::TestCppLibrary::test_mycheck_handler blaPASSED                                                  [ 30%]
tests/test_cpplibrary.py::TestCppLibrary::test_projectionDOFvalues PASSED                                              [ 31%]
tests/test_cpplibrary.py::TestCppLibrary::test_projection_efficiencies PASSED                                          [ 32%]
tests/test_cpplibrary.py::TestCppLibrary::test_rankStructure PASSED                                                    [ 34%]
tests/test_cpplibrary.py::TestCppLibrary::test_splits PASSED                                                           [ 35%]
tests/test_oapackage.py::TestMisc::test_exampleArray PASSED                                                            [ 37%]
tests/test_oapackage.py::TestMisc::test_oa2graph PASSED                                                                [ 38%]
tests/test_oapackage.py::TestMisc::test_reduceGraphNauty PASSED                                                        [ 40%]
tests/test_oapackage.py::TestMisc::test_scanf FAILED                                                                   [ 41%]
tests/test_oapackage.py::test_numpy_interface PASSED                                                                   [ 42%]
tests/test_oapackage.py::test_nauty PASSED                                                                             [ 44%]
tests/test_oapackage.py::TestOAfiles::test_checkArrayFile PASSED                                                       [ 45%]
tests/test_oapackage.py::TestOAfiles::test_finddirectories PASSED                                                      [ 47%]
tests/test_oapackage.py::TestOAfiles::test_findfiles PASSED                                                            [ 48%]
tests/test_oapackage.py::TestOAfiles::test_findfilesR PASSED                                                           [ 50%]
tests/test_oapackage.py::TestOAfiles::test_misc_file_operations PASSED                                                 [ 51%]
tests/test_oapackage.py::TestOAfiles::test_nArrayFile PASSED                                                           [ 52%]
tests/test_oapackage.py::TestOAfiles::test_selectArrays PASSED                                                         [ 54%]
tests/test_oapackage.py::TestParetoFunctionality::test_selectParetoArrays FAILED                                       [ 55%]
tests/test_oapackage.py::TestOAhelper::test_argsort PASSED                                                             [ 57%]
tests/test_oapackage.py::TestOAhelper::test_array2html PASSED                                                          [ 58%]
tests/test_oapackage.py::TestOAhelper::test_array2latex PASSED                                                         [ 60%]
tests/test_oapackage.py::TestOAhelper::test_bounds PASSED                                                              [ 61%]
tests/test_oapackage.py::TestOAhelper::test_checkFiles PASSED                                                          [ 62%]
tests/test_oapackage.py::TestOAhelper::test_checkFilesOA PASSED                                                        [ 64%]
tests/test_oapackage.py::TestOAhelper::test_choose PASSED                                                              [ 65%]
tests/test_oapackage.py::TestOAhelper::test_create_pareto_element PASSED                                               [ 67%]
tests/test_oapackage.py::TestOAhelper::test_deprecated FAILED                                                          [ 68%]
tests/test_oapackage.py::TestOAhelper::test_designStandardError FAILED                                                 [ 70%]
tests/test_oapackage.py::TestOAhelper::test_fac PASSED                                                                 [ 71%]
tests/test_oapackage.py::TestOAhelper::test_floatformat PASSED                                                         [ 72%]
tests/test_oapackage.py::TestOAhelper::test_formatC PASSED                                                             [ 74%]
tests/test_oapackage.py::TestOAhelper::test_gwlp2str PASSED                                                            [ 75%]
tests/test_oapackage.py::TestOAhelper::test_joinArrayLists PASSED                                                      [ 77%]
tests/test_oapackage.py::TestOAhelper::test_parseProcessingTime PASSED                                                 [ 78%]
tests/test_oapackage.py::TestOAhelper::test_plot2Dline PASSED                                                          [ 80%]
tests/test_oapackage.py::TestOAhelper::test_runExtend PASSED                                                           [ 81%]
tests/test_oapackage.py::TestOAhelper::test_safemin_safemax PASSED                                                     [ 82%]
tests/test_oapackage.py::TestOAhelper::test_sortcols PASSED                                                            [ 84%]
tests/test_oapackage.py::TestOAhelper::test_sortrows PASSED                                                            [ 85%]
tests/test_oapackage.py::TestOAhelper::test_testHtml PASSED                                                            [ 87%]
tests/test_oapackage.py::TestDoptimize::test_array2Dtable PASSED                                                       [ 88%]
tests/test_oapackage.py::TestDoptimize::test_calcScore PASSED                                                          [ 90%]
tests/test_oapackage.py::TestDoptimize::test_custom_optim PASSED                                                       [ 91%]
tests/test_oapackage.py::TestDoptimize::test_filterPareto PASSED                                                       [ 92%]
tests/test_oapackage.py::TestDoptimize::test_generateDpage PASSED                                                      [ 94%]
tests/test_oapackage.py::TestDoptimize::test_generateDscatter PASSED                                                   [ 95%]
tests/test_oapackage.py::TestDoptimize::test_optimDeffPython PASSED                                                    [ 97%]
tests/test_oapackage.py::TestDoptimize::test_runcommand PASSED                                                         [ 98%]
tests/test_oapackage.py::TestDoptimize::test_unittest PASSED                                                           [100%]

========================================================== FAILURES ==========================================================
___________________________________________ TestCppLibrary.test_exception_handling ___________________________________________

self = <oapackage.tests.test_cpplibrary.TestCppLibrary testMethod=test_exception_handling>

    def test_exception_handling(self):
        with self.assertRaises(RuntimeError):
            oapackage.mycheck_handler("file", "function", 10, 0, "hi")
        oapackage.mycheck_handler("file", "function", 10, 1, "hi")
    
        with self.assertRaises(RuntimeError):
            oapackage.throw_runtime_exception("dsfs")
    
        al = oapackage.oalib.exampleArray(18, 1)
>       with self.assertRaisesRegex(RuntimeError, "array cannot have negative elements"):
E       AttributeError: 'TestCppLibrary' object has no attribute 'assertRaisesRegex'

tests/test_cpplibrary.py:240: AttributeError
---------------------------------------------------- Captured stdout call ----------------------------------------------------
mycheck: file: function (line 10): exampleArray: conference matrix of size 16, 7
____________________________________________________ TestMisc.test_scanf _____________________________________________________

self = <oapackage.tests.test_oapackage.TestMisc testMethod=test_scanf>

    def test_scanf(self):
>       r = oapackage.scanf.sscanf('1', '%d')

tests/test_oapackage.py:53: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

inputString = '1', formatString = '%d'

    def sscanf(inputString, formatString):
        """ Simple scanf function """
>       return _scanfpip.scanf(formatString, inputString)
E       AttributeError: 'module' object has no attribute 'scanf'

scanf.py:6: AttributeError
______________________________________ TestParetoFunctionality.test_selectParetoArrays _______________________________________

self = <oapackage.tests.test_oapackage.TestParetoFunctionality instance at 0x7ffb0214b200>

    def test_selectParetoArrays(self):
    
        arrays = [oapackage.array_link(np.array([[ii]])) for ii in range(5)]
        pareto_object = oapackage.ParetoLongLong()
    
        for ii in range(len(arrays)):
            value = [ii, ii % 2]
            pareto_object.addvalue(value, ii)
        pareto_object.show(2)
    
        selected = oapackage.oahelper.selectParetoArrays(arrays, pareto_object)
>       self.assertEqual(selected, arrays[4, 5])
E       AttributeError: TestParetoFunctionality instance has no attribute 'assertEqual'

tests/test_oapackage.py:241: AttributeError
---------------------------------------------------- Captured stdout call ----------------------------------------------------
Pareto: 2 optimal values, 2 objects
value 0: 3; 1
value 1: 4; 0
________________________________________________ TestOAhelper.test_deprecated ________________________________________________

self = <oapackage.tests.test_oapackage.TestOAhelper testMethod=test_deprecated>

    def test_deprecated(self):
        def func():
            return 'hi'
        deprecated_function = oapackage.oahelper.deprecated(func)
>       with self.assertWarns(Warning):
E       AttributeError: 'TestOAhelper' object has no attribute 'assertWarns'

tests/test_oapackage.py:290: AttributeError
---------------------------------------------------- Captured stdout call ----------------------------------------------------
exampleArray: array 3 in OA(16, 2, 2^5)
___________________________________________ TestOAhelper.test_designStandardError ____________________________________________

self = <oapackage.tests.test_oapackage.TestOAhelper testMethod=test_designStandardError>

    def test_designStandardError(self):
        al = oapackage.exampleArray(14, 0)
        v = oapackage.oahelper.designStandardError(al)
>       self.assertAlmostEqual(v[0], 0.3747931073686535)
E       AssertionError: 0.2288887948318749 != 0.3747931073686535 within 7 places

tests/test_oapackage.py:363: AssertionError
---------------------------------------------------- Captured stdout call ----------------------------------------------------
exampleArray: array 3 in OA(16, 2, 2^5)
====================================================== warnings summary ======================================================
tests/test_oapackage.py::TestOAhelper::test_designStandardError
  /usr/local/lib/python2.7/dist-packages/oapackage/oahelper.py:1167: RuntimeWarning: invalid value encountered in sqrt
    return np.sqrt(m0), np.sqrt(m1), np.sqrt(m2)

None
  could not create cache path /usr/local/lib/python2.7/dist-packages/oapackage/.cache/v/cache/lastfailed

-- Docs: http://doc.pytest.org/en/latest/warnings.html
====================================== 5 failed, 65 passed, 2 warnings in 0.81 seconds =======================================

**Comment by eendebakpt** [comment]
@gsagnol About the first issue (on your PC). It seems the python 2.7 and 3.6.5 distrubutions are mixed up. Since I have no access to your system, it is hard for me to explain how this could happen.

I will try to make a few suggestions. In the output of `pytest -v` it reads
```
...
ImportError while importing test module '/homes/-hidden-/.local/lib/python2.7/site-packages/oapackage/tests/test_cpplibrary.py'
...
./oalib.py:19: in swig_import_helper
return importlib.import_module('_oalib')
/usr/lib64/python3.6/importlib/init.py:126: in import_module
...
```
The first which means that the `oapackage` was installed (and compiled) as version 2.7.  The last lines show that `oalib.py` tries to use the python 3.6 `importlib`.

It appears you are running linux (or at least some unix variation). Are there besides `pytest` also `pytest-2`  and `pytest-2` available. If so, can you run `pytest-2`?

Can you also try to create a [virtualenv ](https://virtualenv.pypa.io/en/latest/) environment and the the oapackage there?

**Comment by eendebakpt** [comment]
@gsagnol In the second report there seem to be multiple things going on. Some of them I fixed in #96, but for the others I need some more information. Can you run for me the following python script and send me the output:
```
import numpy as np; np.show_config()

import oapackage
al = oapackage.exampleArray(14, 1)
al.showarray()
print('Defficiencies: %s' % (al.Defficiencies(), ) )
              
scalefac=1
X=al.getModelMatrix(2)
print('model matrix:')
print(X)
M = np.linalg.inv(X.transpose().dot(X) / scalefac)
print('information matrix:')
print(M)
v = oapackage.oahelper.designStandardError(al)
print('output of designStandardError')
print(v)
print(v[0])
```
And also the following script:
```
import unittest
print(unittest)
print(dir(unittest.TestCase))
print(unittest.TestCase.assertWarns)
```

**Comment by gsagnol** [comment]
Hi ,

You're correct the python installation of my PC's institute is mixed up. Since I am not administrating it, the simplest solution would indeed be to use virtualenvs, for now I will focus on the results I obtain on my laptop.

**Comment by gsagnol** [comment]
Here is the ouput of your first script

Python 2.7.15rc1 (default, Nov 12 2018, 14:31:15) 
Type "copyright", "credits" or "license" for more information.

IPython 5.5.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.
Using matplotlib backend: TkAgg

In [1]: import numpy as np; np.show_config()
   ...: 
   ...: import oapackage
   ...: al = oapackage.exampleArray(14, 1)
   ...: al.showarray()
   ...: print('Defficiencies: %s' % (al.Defficiencies(), ) )
   ...:               
   ...: scalefac=1
   ...: X=al.getModelMatrix(2)
   ...: print('model matrix:')
   ...: print(X)
   ...: M = np.linalg.inv(X.transpose().dot(X) / scalefac)
   ...: print('information matrix:')
   ...: print(M)
   ...: v = oapackage.oahelper.designStandardError(al)
   ...: print('output of designStandardError')
   ...: print(v)
   ...: print(v[0])
   ...: 
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    define_macros = [('HAVE_CBLAS', None)]
    language = c
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    define_macros = [('HAVE_CBLAS', None)]
    language = c
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    define_macros = [('HAVE_CBLAS', None)]
    language = c
blis_info:
  NOT AVAILABLE
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    define_macros = [('HAVE_CBLAS', None)]
    language = c
lapack_mkl_info:
  NOT AVAILABLE
blas_mkl_info:
  NOT AVAILABLE
exampleArray: design in D(28, 2^5), D-efficiency is low
array: 
  1   0   1   1   1
  1   0   1   0   0
  0   1   0   1   1
  0   0   1   0   1
  1   1   1   0   0
  0   0   0   0   0
  0   1   0   1   1
  1   0   1   0   0
  1   0   0   0   0
  1   0   0   1   1
  0   1   1   0   0
  0   1   0   1   1
  1   0   1   1   0
  1   0   1   0   1
  0   0   1   0   0
  0   1   0   0   0
  0   0   0   1   1
  0   1   0   0   1
  0   1   0   1   1
  1   1   0   0   0
  1   0   1   0   1
  1   1   0   0   1
  1   1   0   1   0
  0   0   1   0   1
  0   1   0   1   0
  1   1   0   0   0
  0   1   1   0   0
  1   0   1   0   1
Defficiencies: (0.0, 0.0, 0.8727271801272938)
model matrix:
[[ 1.  1. -1.  1.  1.  1. -1.  1.  1.  1. -1. -1. -1.  1.  1.  1.]
 [ 1.  1. -1.  1. -1. -1. -1.  1. -1. -1. -1.  1.  1. -1. -1.  1.]
 [ 1. -1.  1. -1.  1.  1. -1.  1. -1. -1. -1.  1.  1. -1. -1.  1.]
 [ 1. -1. -1.  1. -1.  1.  1. -1.  1. -1. -1.  1. -1. -1.  1. -1.]
 [ 1.  1.  1.  1. -1. -1.  1.  1. -1. -1.  1. -1. -1. -1. -1.  1.]
 [ 1. -1. -1. -1. -1. -1.  1.  1.  1.  1.  1.  1.  1.  1.  1.  1.]
 [ 1. -1.  1. -1.  1.  1. -1.  1. -1. -1. -1.  1.  1. -1. -1.  1.]
 [ 1.  1. -1.  1. -1. -1. -1.  1. -1. -1. -1.  1.  1. -1. -1.  1.]
 [ 1.  1. -1. -1. -1. -1. -1. -1. -1. -1.  1.  1.  1.  1.  1.  1.]
 [ 1.  1. -1. -1.  1.  1. -1. -1.  1.  1.  1. -1. -1. -1. -1.  1.]
 [ 1. -1.  1.  1. -1. -1. -1. -1.  1.  1.  1. -1. -1. -1. -1.  1.]
 [ 1. -1.  1. -1.  1.  1. -1.  1. -1. -1. -1.  1.  1. -1. -1.  1.]
 [ 1.  1. -1.  1.  1. -1. -1.  1.  1. -1. -1. -1.  1.  1. -1. -1.]
 [ 1.  1. -1.  1. -1.  1. -1.  1. -1.  1. -1.  1. -1. -1.  1. -1.]
 [ 1. -1. -1.  1. -1. -1.  1. -1.  1.  1. -1.  1.  1. -1. -1.  1.]
 [ 1. -1.  1. -1. -1. -1. -1.  1.  1.  1. -1. -1. -1.  1.  1.  1.]
 [ 1. -1. -1. -1.  1.  1.  1.  1. -1. -1.  1. -1. -1. -1. -1.  1.]
 [ 1. -1.  1. -1. -1.  1. -1.  1.  1. -1. -1. -1.  1.  1. -1. -1.]
 [ 1. -1.  1. -1.  1.  1. -1.  1. -1. -1. -1.  1.  1. -1. -1.  1.]
 [ 1.  1.  1. -1. -1. -1.  1. -1. -1. -1. -1. -1. -1.  1.  1.  1.]
 [ 1.  1. -1.  1. -1.  1. -1.  1. -1.  1. -1.  1. -1. -1.  1. -1.]
 [ 1.  1.  1. -1. -1.  1.  1. -1. -1.  1. -1. -1.  1.  1. -1. -1.]
 [ 1.  1.  1. -1.  1. -1.  1. -1.  1. -1. -1.  1. -1. -1.  1. -1.]
 [ 1. -1. -1.  1. -1.  1.  1. -1.  1. -1. -1.  1. -1. -1.  1. -1.]
 [ 1. -1.  1. -1.  1. -1. -1.  1. -1.  1. -1.  1. -1. -1.  1. -1.]
 [ 1.  1.  1. -1. -1. -1.  1. -1. -1. -1. -1. -1. -1.  1.  1.  1.]
 [ 1. -1.  1.  1. -1. -1. -1. -1.  1.  1.  1. -1. -1. -1. -1.  1.]
 [ 1.  1. -1.  1. -1.  1. -1.  1. -1.  1. -1.  1. -1. -1.  1. -1.]]
information matrix:
[[ 5.23900804e-02  4.40546476e-02  1.67294674e-02 -1.09175849e-01
  -1.48444038e-01  2.23386805e-02 -5.74357211e-02 -1.57955342e-01
   9.89626920e-02  1.50432120e-01 -7.70154665e-02 -6.07261281e+12
  -1.02114117e+15  2.37004064e+15  6.07261281e+12  1.03511731e-02]
 [-3.00539040e-01  7.91325583e-02  2.17834688e-02  5.55470361e-02
  -1.12884746e-01  2.98224844e-02  8.38313238e-03 -9.94862751e-03
   1.67319989e-01  5.67516911e-02 -9.06625238e-02  1.10820717e+15
  -2.38445672e+15  6.29712244e+14 -1.10820717e+15 -1.12712774e-02]
 [-2.51790796e-01  8.32877541e-02  1.90154884e-03  1.73068207e-01
  -1.41259009e-01  4.20598409e-02 -3.56481703e-02  2.53987119e-02
   2.49728228e-01  2.15668183e-01 -6.34812316e-02 -1.25687025e+14
  -2.74203216e+15  1.75667167e+15  1.25687025e+14  6.92378470e-02]
 [-1.66538874e-01  6.67650667e-02 -6.57874559e-02  7.15381077e-02
  -2.78216531e-01  5.02415274e-02 -6.81625949e-02  3.86252109e-02
   1.85477687e-01  5.60496856e-02 -2.78216531e-01  3.65343383e+14
  -1.23553745e+15  2.10573152e+15 -3.65343383e+14  1.44916424e-01]
 [ 8.94189110e-02  1.14838595e-02 -4.20368210e-02 -1.09903939e-01
  -2.40559882e-01  6.16232411e-03 -4.42429975e-02  1.08906387e-01
   3.39096208e-02 -6.78269211e-02 -1.56376473e-01  1.23938186e+14
  -4.95973065e+14  2.32908627e+15 -1.23938186e+14  8.48049897e-02]
 [ 1.36405808e-01  1.69138385e-02  5.27963944e-02  1.50032610e-01
  -7.01375452e-03  6.31803936e-02 -1.47972431e-03 -2.86275014e-02
   1.63007471e-02  1.21908990e-01  5.36394405e-02 -1.82781649e+14
  -1.00885820e+15  3.07491837e+14  1.82781649e+14  7.05747331e-05]
 [-1.69750291e-01  1.11327819e-02  1.32860823e-01  2.15464549e-01
  -1.26764511e-01 -4.08925699e-03  7.58802084e-02 -1.00631707e-01
  -1.36079609e-02  7.11941059e-02 -1.92882410e-03 -2.27405083e+14
   2.04778756e+13  7.60300676e+14  2.27405083e+14 -1.31388903e-01]
 [ 5.61039303e-01 -2.57272095e-02  8.00618769e-02  4.41731045e-01
   1.68131520e-01 -1.02032252e-02  6.76634297e-02 -1.16318801e-01
  -1.22789383e-01  1.55109319e-01  3.94263661e-01 -4.07319378e+14
  -8.75712038e+13 -1.36798933e+15  4.07319378e+14 -1.03423797e-01]
 [ 4.67550345e-02  1.10140760e-01 -2.11227201e-02  1.85914078e-02
  -1.66347320e-01  3.01582477e-02 -3.52045335e-03  4.70315497e-02
   1.89724837e-01  1.08805975e-01  6.56229574e-02  1.31993138e+15
  -3.23938675e+15  5.37350460e+14 -1.31993138e+15  2.81636268e-02]
 [-1.58590521e-01  1.98238151e-02  6.78479236e-02 -6.81266226e-02
  -2.14991107e-01 -4.25791391e-03  1.13079873e-02 -1.46864485e-01
   1.35695847e-01  1.67529567e-02 -2.14991107e-01 -8.65125428e+14
   3.57114105e+14  1.62965271e+15  8.65125428e+14 -9.04638981e-02]
 [ 4.62970717e-01 -5.78713396e-02  0.00000000e+00  4.62970717e-01
   2.31485359e-01  2.89356698e-02  0.00000000e+00 -2.31485359e-01
   0.00000000e+00  3.47228038e-01  2.31485359e-01 -1.04251737e+15
  -1.04251737e+15  0.00000000e+00  1.04251737e+15  0.00000000e+00]
 [ 5.97232515e+15 -1.04284550e+15  1.28274892e+15  4.80154535e+15
   2.74402402e+15 -3.48963336e+14  6.50097657e+14 -2.26528152e+15
  -2.39321582e+15  1.55167349e+15  3.19282295e+15 -1.49349093e+31
   1.49349093e+31 -1.49349093e+31  1.49349093e+31 -2.25179981e+15]
 [-5.97232515e+15  1.04284550e+15 -1.28274892e+15 -4.80154535e+15
  -2.74402402e+15  3.48963336e+14 -6.50097657e+14  2.26528152e+15
   2.39321582e+15 -1.55167349e+15 -3.19282295e+15  1.49349093e+31
  -1.49349093e+31  1.49349093e+31 -1.49349093e+31  2.25179981e+15]
 [ 5.97232515e+15 -1.04284550e+15  1.28274892e+15  4.80154535e+15
   2.74402402e+15 -3.48963336e+14  6.50097657e+14 -2.26528152e+15
  -2.39321582e+15  1.55167349e+15  3.19282295e+15 -1.49349093e+31
   1.49349093e+31 -1.49349093e+31  1.49349093e+31 -2.25179981e+15]
 [-5.97232515e+15  1.04284550e+15 -1.28274892e+15 -4.80154535e+15
  -2.74402402e+15  3.48963336e+14 -6.50097657e+14  2.26528152e+15
   2.39321582e+15 -1.55167349e+15 -3.19282295e+15  1.49349093e+31
  -1.49349093e+31  1.49349093e+31 -1.49349093e+31  2.25179981e+15]
 [ 1.50000000e-01 -1.25000000e-02  5.00000000e-02  1.50000000e-01
   7.50000000e-02  1.25000000e-02  2.77343750e-02 -6.25000000e-02
  -5.00000000e-02  5.00000000e-02  7.50000000e-02 -4.50359963e+14
   4.50359963e+14 -4.50359963e+14  4.50359963e+14  0.00000000e+00]]
/usr/local/lib/python2.7/dist-packages/oapackage/oahelper.py:1167: RuntimeWarning: invalid value encountered in sqrt
  return np.sqrt(m0), np.sqrt(m1), np.sqrt(m2)
output of designStandardError
(0.2288887948318749, array([       nan, 0.04360675, 0.2513571 , 0.26746609, 0.2813051 ]), array([       nan,        nan,        nan,        nan,        nan,
       0.        , 0.12943321, 0.27546362, 0.43557415, 0.48112925]))
0.228888794832

**Comment by gsagnol** [comment]
There seems to be a number of entries of magnitude > 1e15 in the inverse of the information matrix, is this normal?

**Comment by gsagnol** [comment]
And here is the output of my second script: I guess I have to update `unittest` ?
```
In [2]: import unittest
   ...: print(unittest)
   ...: print(dir(unittest.TestCase))
   ...: print(unittest.TestCase.assertWarns)
   ...: 
<module 'unittest' from '/usr/lib/python2.7/unittest/__init__.pyc'>
['__call__', '__class__', '__delattr__', '__dict__', '__doc__', '__eq__', '__format__', '__getattribute__', '__hash__', '__init__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_addSkip', '_baseAssertEqual', '_classSetupFailed', '_deprecate', '_diffThreshold', '_formatMessage', '_getAssertEqualityFunc', '_truncateMessage', 'addCleanup', 'addTypeEqualityFunc', 'assertAlmostEqual', 'assertAlmostEquals', 'assertDictContainsSubset', 'assertDictEqual', 'assertEqual', 'assertEquals', 'assertFalse', 'assertGreater', 'assertGreaterEqual', 'assertIn', 'assertIs', 'assertIsInstance', 'assertIsNone', 'assertIsNot', 'assertIsNotNone', 'assertItemsEqual', 'assertLess', 'assertLessEqual', 'assertListEqual', 'assertMultiLineEqual', 'assertNotAlmostEqual', 'assertNotAlmostEquals', 'assertNotEqual', 'assertNotEquals', 'assertNotIn', 'assertNotIsInstance', 'assertNotRegexpMatches', 'assertRaises', 'assertRaisesRegexp', 'assertRegexpMatches', 'assertSequenceEqual', 'assertSetEqual', 'assertTrue', 'assertTupleEqual', 'assert_', 'countTestCases', 'debug', 'defaultTestResult', 'doCleanups', 'fail', 'failIf', 'failIfAlmostEqual', 'failIfEqual', 'failUnless', 'failUnlessAlmostEqual', 'failUnlessEqual', 'failUnlessRaises', 'failureException', 'id', 'longMessage', 'maxDiff', 'run', 'setUp', 'setUpClass', 'shortDescription', 'skipTest', 'tearDown', 'tearDownClass']
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-2-fa12ea56c967> in <module>()
      2 print(unittest)
      3 print(dir(unittest.TestCase))
----> 4 print(unittest.TestCase.assertWarns)

AttributeError: type object 'TestCase' has no attribute 'assertWarns'
```

**Comment by peendebak** [comment]
@gsagnol The errors have been fixed in #105, no update of `unittest` should be necessary. The entries in the information matrix are large (so is the condition number of M), but this is fine since the input array is a special case used for testing (in the output you see `exampleArray: design in D(28, 2^5), D-efficiency is low`).
